Group our transactions[WIP]

RokoVarano opened the pull request:
This is a work in progress of the Ruby on Rails capstone project. It is not done yet.
Currently present:
- Login and Signup homepage, where a user can log in with his name alone.
Validations include uniqueness and existance of name
- Navbar that includes homepage link, log out link, current page title and flash notifications in all pages except Login
- Footer for "New Item" button, including Tasks and Groups. Appear only in Task and Groups views
- User profile page with his name and avatar. The avatar can be changed by uploading a file.
- Links to "All Tasks", "External Tasks", and "Groups"
- Tasks view listing all tasks or non-group tasks created by the current user, depending on the link clicked. Group associated tasks include the picture of the first group they are associated with.
- New task view, with name and points fields
- Groups view listing all current-user created groups. Includes picture of the group, name, and creation date.
- When clicking a group, a Group-Task association page shows, where tasks can be added to the group.

Work will continue to be done next week. Apologies for the long time it has taken to complete this project.

Thanks in advance for your patience and disposition.

RokoVarano committed:
gitflow-feature-stash: workspace

RokoVarano committed:
create login

RokoVarano committed:
style login page

RokoVarano committed:
Login and Signup in the same page

RokoVarano committed:
Create Task

RokoVarano committed:
Create Group and group relationships

RokoVarano committed:
Fix group and Tasks relationships

RokoVarano committed:
Logged in users are redirected from Login page

RokoVarano committed:
Create navbar

RokoVarano committed:
User bar, Total tasks bar

RokoVarano committed:
Add tasks to groups in group view

RokoVarano committed:
Style tasks

RokoVarano committed:
Style footer

RokoVarano committed:
Fix login crash

RokoVarano committed:
Test Login/Create homepage

RokoVarano committed:
Task tests

RokoVarano committed:
Change Ruby version for Lintern compatibility

RokoVarano committed:
Rubocop corrections

RokoVarano committed:
Fix cyclomatic complexity of navbar_title

RokoVarano committed:
Fix additional rubocop errors

henatan99 reviewed (commented):
- There is a clear difference between the original design and that of your app. Please have a look at the attached screenshot images to put them into perspective. 

![Screenshot from 2021-05-01 04-08-37](https://user-images.githubusercontent.com/64894927/116762194-5238db80-aa22-11eb-8e78-0b4b9e4acbd9.png)
![picture1](https://user-images.githubusercontent.com/64894927/116762223-667cd880-aa22-11eb-819d-288b4635aef7.jpg)

RokoVarano committed:
Task widget styling

RokoVarano committed:
Remove footer from incorrect views, style group#show view

RokoVarano committed:
Footer styling

RokoVarano committed:
Add unit tests, rubocop -a

RokoVarano commented:
I am asking in this review to check for functionality and style impressions, before doing the rest of the app. 
As personalization, instead of using money transactions, this app is aimed at task management for roommantes. Each task can be given points. For example, cleaning the toilet is gross, so a user can give it 8 points, while cooking gets only 5 points. 
Because User can be logged and created only with the name, it makes sense that Login and Sign up share a single screen. This is part of my personalization. Picture can be changed in the profile page.

- Style has been adapted to look more like the examples
- Unit and integration tests have been added

To do:
- Upload to Heroku
- change DB to Postgres
- Dry the code in general

RokoVarano committed:
Add relational tests

RokoVarano committed:
Add task to group on creation, order tasks#index by date

RokoVarano committed:
Update profile page

RokoVarano committed:
Protect routes, change tests to accomodate for new profile page

RokoVarano committed:
Move 'tasks#index' logic to model

RokoVarano committed:
Add user_name validations

RokoVarano committed:
Filter tasks with groups in 'My Tasks' page

RokoVarano committed:
Solve n+1 problem with Bullet

RokoVarano committed:
Disable Bullet

RokoVarano committed:
Clear Lintern errors

RokoVarano committed:
Change Ruby Version

RokoVarano commented:
- Changed Design to look more like the original template
- Added validation, integration and relational tests
- Checked and solved n +1 problems
- Removed logic from Task Controller
- Solved mixed tasks showing in 'My tasks' page
- Tasks can be included into a group at the moment of creation
- Tasks and groups are always in the requiered corresponding order

In this request I ask you to check for issues before uploading to Heroku and doing the video, since those steps should be done only when the project is ready for it.

RokoVarano committed:
Add feature: 3 best colaborators in group

RokoVarano committed:
Change image storage to Amazon

RokoVarano committed:
Save missing changes to AWS config

RokoVarano committed:
Update readme.md

RokoVarano committed:
Set Postgresql as DB

RokoVarano committed:
Last touches
